test/CurveFactoryV2.t.sol

```diff
@@ -172,4 +168,94 @@ contract CurveFactoryV2Test is Test {
         cheats.prank(address(liquidityProvider));
         dfxCadcCurve.withdraw(100_000e18, block.timestamp + 60);
     }
+
+    function test_depositGlobalGuard() public {
+        // enable global guard
+        curveFactory.toggleGlobalGuarded();
+        // set global guard amount to 100k
+        curveFactory.setGlobalGuardAmount(100_000e18);
+
+        deal(address(cadc), address(liquidityProvider), 200_000e18);
+        deal(address(usdc), address(liquidityProvider), 200_000e6);
+
+        cheats.startPrank(address(liquidityProvider));
+        cadc.approve(address(dfxCadcCurve), type(uint).max);
+        usdc.approve(address(dfxCadcCurve), type(uint).max);
+
+        dfxCadcCurve.deposit(100_000e18, block.timestamp + 60);
+        cheats.stopPrank();
+    }
+
+    function testFail_depositGlobalGuard() public {
+
+        // enable global guard
+        curveFactory.toggleGlobalGuarded();
+        // set global guard amount to 100k
+        curveFactory.setGlobalGuardAmount(100_000e18);
+
+        deal(address(cadc), address(liquidityProvider), 200_000e18);
+        deal(address(usdc), address(liquidityProvider), 200_000e6);
+
+        cheats.startPrank(address(liquidityProvider));
+        cadc.approve(address(dfxCadcCurve), type(uint).max);
+        usdc.approve(address(dfxCadcCurve), type(uint).max);
+
+        dfxCadcCurve.deposit(110_000e18, block.timestamp + 60);
+        cheats.stopPrank();
+    }
+
+    function test_depositPoolGuard() public {
+        
+        // enable global guard
+        curveFactory.toggleGlobalGuarded();
+        // set global guard amount to 100k
+        curveFactory.setGlobalGuardAmount(100_000e18);
+        // while global guard amt is 100k, Euroc pool guard amt is 80k
```

Comment:
Can you add a test to make sure that values within > 80k but < 100k doesn't work?